kernel/pci/src/lib.rs

```diff
@@ -221,74 +263,98 @@ impl PciLocation {
     /// The least two significant bits of offset are masked, so it's 4-byte aligned addressing,
     /// which makes sense since we read PCI data (from the configuration space) in 32-bit chunks.
     fn pci_address(self, offset: u8) -> u32 {
-        ((self.bus  as u32) << 16) | 
-        ((self.slot as u32) << 11) | 
-        ((self.func as u32) <<  8) | 
-        ((offset as u32) & (PCI_CONFIG_ADDRESS_OFFSET_MASK as u32)) | 
-        0x8000_0000
+        BASE_OFFSET |
+        ((self.bus  as u32) << 16) |
+        ((self.slot as u32) << 11) |
+        ((self.func as u32) <<  8) |
+        (offset as u32)
```

Comment:
here you're no longer bit-masking out the least two significant bits. Is that correct? It doesn't seem right.

Reply:
Indeed; I moved this to the read & write methods